hazelcast-tpc-engine/src/main/java/com/hazelcast/internal/tpc/nio/NioAcceptRequest.java

```diff
@@ -20,15 +20,13 @@
 
 import java.nio.channels.SocketChannel;
 
-public class NioAcceptRequest implements AcceptRequest {
+import static com.hazelcast.internal.tpc.util.Preconditions.checkNotNull;
 
-    private final SocketChannel socketChannel;
+class NioAcceptRequest implements AcceptRequest {
 
-    public NioAcceptRequest(SocketChannel socketChannel) {
-        this.socketChannel = socketChannel;
-    }
+    final SocketChannel socketChannel;
```

Comment:
I think using getter instead of package private field is better.

Reply:
Package private reduces unwanted coupling. Nobody needs to know what kind of object is being passed.

Reply:
Keep in mind that only the nio package creates this NioAcceptRequest and only this package consumes it.